test/libsolidity/semanticTests/calldata/calldata_bound_dynamic_array_or_slice.sol

```diff
@@ -0,0 +1,25 @@
+pragma abicoder v2;
+
+library L {
+    function reverse(uint[] calldata _a) internal pure returns (uint, uint) {
+        return (_a[1], _a[0]);
+    }
+}
+
+contract C {
+    using L for *;
+
+    function testArray(uint, uint[] calldata _a, uint) external pure returns (uint, uint) {
+        return _a.reverse();
+    }
+
+    function testSlice(uint, uint[] calldata _a, uint) external pure returns (uint, uint) {
+        return _a[:].reverse();
+    }
+}
+
+// ====
+// compileViaYul: also
+// ----
+// testArray(uint256,uint256[],uint256): 7, 0x60, 4, 2, 66, 77 -> 77, 66
+// testSlice(uint256,uint256[],uint256): 7, 0x60, 4, 1, 66, 77 -> FAILURE, hex"4e487b71", 0x32
```

Comment:
I have added a slice in this test. Curiously, it reverts even though array does not. Is this the expected behavior?

Reply:
The input array is only 1 element long (the one above is 2 long), so `a[:]` also only has length `1` but in `reverse` you access `_a[1]`

Reply:
Ah, right. Thanks! So it's my mistake.

Reply:
Fixed. Now it returns 77, 66 as expected.